### ISSUE TITLE
TypeError: Cannot read property '_ownerDocument' of undefined

### ISSUE THREAD

**Issue by cimbis** [jsdom/jsdom#3121]
I've been fruitlessly trying to set up vanilla JS / HTML tests. 

Example code provided below.

Full error message:

```
TypeError: Cannot read property '_ownerDocument' of undefined

    at innerInvokeEventListeners (/Users/jc/Desktop/JSDOM/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:327:26)
    at invokeEventListeners (/Users/jc/Desktop/JSDOM/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:274:3)
    at EventTargetImpl._dispatch (/Users/jc/Desktop/JSDOM/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:221:9)
    at fireAnEvent (/Users/jc/Desktop/JSDOM/node_modules/jsdom/lib/jsdom/living/helpers/events.js:18:36)
    at /Users/jc/Desktop/JSDOM/node_modules/jsdom/lib/jsdom/browser/Window.js:773:7
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
```

### Basic info:

- **Node.js version: v12.16.3**
- **jsdom version: v16.4.0**

### Minimal reproduction case

https://github.com/cimbis/JSDOM / https://github1s.com/cimbis/JSDOM

Any pointers to the issue I might have made would be very appreciated.

Thanks!

**Comment by domenic** [comment]
Can you create a *minimal* reproduction case involving no third-party libraries?

**Comment by cimbis** [comment]
@domenic updated the repository with JSDOM test involving only built in node packages and JSDOM.

**Comment by domenic** [comment]
This might be fixed in v16.15.0 as we discovered another test case. I haven't yet tested yours though.

**Comment by MikePikanin** [comment]
Also have this kind of issue sometimes, but for me it comes from JSDOMParse5Adapter - though it works for me in v15.2.1
Tried to use with parse5 v5.1.0 package (which was in v15.2.1) but have same error.

```
TypeError: Cannot read property '_ownerDocument' of undefined
    at JSDOMParse5Adapter._ownerDocument (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/parser/html.js:64:33)
    at JSDOMParse5Adapter.createElement (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/parser/html.js:86:32)
    at Parser._insertElement (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:559:42)
    at startTagInForeignContent (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:2935:15)
    at Parser._processTokenInForeignContent (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:674:13)
    at Parser._processInputToken (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:682:18)
    at Parser._runParsingLoop (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:440:18)
    at Parser.parse (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/parser/index.js:344:14)
    at Object.parse (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/parse5/lib/index.js:10:19)
    at parseIntoDocument (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/parser/html.js:216:17)
```

Note: for 16.0.0-16.2.2 versions I got another error:
```
TypeError: Cannot read property 'some' of undefined
    at Object.exports.install (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/node_modules/whatwg-url/dist/URL.js:84:20)
    at exports.installInterfaces (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/living/interfaces.js:202:24)
    at setupWindow (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/Window.js:63:3)
    at new Window (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/Window.js:107:3)
    at exports.createWindow (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/jsdom/browser/Window.js:38:10)
    at new JSDOM (/var/lib/nave/global/12.18.3/lib/node_modules/jsdom/lib/api.js:36:20)
    at parseDOM (/home/mikhailp/Downloads/tst.js:35:37)
    at Object.convert (/var/lib/nave/global/12.18.3/lib/node_modules/mhtml2html/dist/mhtml2html.js:1:4773)
    at /home/mikhailp/Downloads/tst.js:33:36
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:63:3)
```

**Comment by domenic** [comment]
@cimbis I can confirm the error is fixed in v16.15.1! I'll close this issue. Thanks for your great reproduction case, again.

@MikePikanin your example of just HTML doesn't show how to trigger the error using Node.js and jsdom. I encourage you to create an example using https://npm.runkit.com/jsdom or a separate repo like @cimbis did, and open a new issue that follows the issue template.
